xmtp_mls/src/groups/mod.rs

```diff
@@ -522,8 +541,44 @@ where
     // Update this installation's leaf key in the group by creating a key update commit
     pub async fn key_update(&self) -> Result<(), GroupError> {
         let conn = &mut self.client.store.conn()?;
+        
+        self.pre_intent_hook().await.unwrap();
+
+        let intent = NewGroupIntent::new(IntentKind::KeyUpdate, self.group_id.clone(), vec![]);
+        intent.store(conn)?;
+
+        self.sync_with_conn(conn).await
+    }
+
+    // Update the installation's leaf key in a synchronous way.
+    pub fn queue_key_update(&self) -> Result<(), GroupError> {
+        let conn = &mut self.client.store.conn()?;
+        
+        self.pre_intent_hook();
+
         let intent = NewGroupIntent::new(IntentKind::KeyUpdate, self.group_id.clone(), vec![]);
         intent.store(conn)?;
+        Ok(())
+    }
+
+    /// Checking the last key rotation time before rotating the key. 
+    pub async fn pre_intent_hook(&self) -> Result<(), GroupError> {
+        let conn = &self.client.store.conn()?;
+        let last_rotated_time = conn.get_rotated_time_checked(self.group_id.clone());
+        
+        match last_rotated_time {
+            Ok(rotated_time) => {
+                if rotated_time == 0 {
+                    let _ = self.queue_key_update();
```

Comment:
The `rotated_time` is not currently updated after the key update is queued. That means the key update will be queued every single time, rather than only on an interval. I wonder if it makes sense to make a method `maybe_queue_key_update` similar to `maybe_update_installations`, that both checks the `rotated_time` and updates it if a key update is queued.

It might make sense to also add a unit test to catch this error